src/Headers/Ic.h

```diff
@@ -104,7 +105,7 @@ class Ic
   virtual void Generate(void) {
     ExceptionHandler::getIstance().raise("Ic::Generate function not implemented");
   };
-  virtual FLOAT GetDensity(const FLOAT *) const {
+  virtual FLOAT GetDensity(const FLOAT *, const Typemask typemask=Typemask(true)) const {
```

Comment:
> t's probably best to give it just the particle type (btw, on a slight tangent, but we have both ptype and parttype. Do we really need both? 

I think the only place the parttype is used is in the IO routines. I think it only still exists as because we never finished removing it. IIRC I originally had to introduce a second variable ptype to make the dust work. We originally had an itype variable as well, that was used for boundaries etc and was typically set to one of the parttype variables, I managed to get rid of itype eventually, so parttype can probably go now too,